Content.Server/Construction/ConstructionSystem.Initial.cs

```diff
@@ -463,78 +481,87 @@ public async Task<bool> TryStartItemConstruction(string prototype, EntityUid use
         // LEGACY CODE. See warning at the top of the file!
         private async void HandleStartStructureConstruction(TryStartStructureConstructionMessage ev, EntitySessionEventArgs args)
         {
-            if (!PrototypeManager.TryIndex(ev.PrototypeName, out ConstructionPrototype? constructionPrototype))
-            {
-                Log.Error($"Tried to start construction of invalid recipe '{ev.PrototypeName}'!");
-                RaiseNetworkEvent(new AckStructureConstructionMessage(ev.Ack));
-                return;
-            }
+            // <Goobstation> - use public API
+            if (args.SenderSession.AttachedEntity is {} user)
+                await TryStartStructureConstruction(user, ev.PrototypeName, GetCoordinates(ev.Location), ev.Angle, ev.Ack, args.SenderSession);
+        }
 
-            if (!PrototypeManager.TryIndex(constructionPrototype.Graph, out ConstructionGraphPrototype? constructionGraph))
+        /// <summary>
+        /// Goobstation - Taken out of HandleStartStructureConstruction
+        /// Changed to return false and only send the ack event to the user.
+        /// </summary>
+        public async Task<bool> TryStartStructureConstruction(EntityUid user, string prototypeName, EntityCoordinates location, Angle angle, int ack = 0, ICommonSession? senderSession = null)
```

Comment:
holy function parameters batman
```suggestion
        public async Task<bool> TryStartStructureConstruction(EntityUid user,
            string prototypeName,
            EntityCoordinates location,
            Angle angle,
            int ack = 0,
            ICommonSession? senderSession = null)

Reply:
johnflict